### ISSUE TITLE
Update pagination to the new approach

### ISSUE THREAD

**Issue by jrambla** [ga4gh-beacon/beacon-framework-v2#9]
https://github.com/ga4gh-beacon/specification-v2/issues/26
https://github.com/ga4gh-beacon/specification-v2/issues/70

**Comment by jrambla** [comment]
@Tom-Shorter @jlfernandez-cba @mbaudis 
After looking to how different APIs address the pagination problem, I've found the Microsoft API documentation to provide an broader consideration (https://github.com/Microsoft/api-guidelines/blob/master/Guidelines.md#98-pagination).
Paypal one is also worth to read (https://github.com/paypal/api-standards/blob/master/patterns.md#pagination)
Their description matches my "concerns":

1. I see the value of using the `next_page` approach. Totally makes sense for long collections, when calculating the page offset  is expensive. I believe that would be find that scenario quite often in the genomics domain.
3. Is that cost also challenging the "numTotalResults"? or it is not?
4. (so far, we have ignored the following aspect) Does pagination *always* imply sorting? If so, should we add sorting now to the spec?
5. The guidelines cited above seem to allow for both solutions (`page `& `next_page`). I'm starting to consider this option, although José Luis suggestion of using the page number in `next_page` could solve that.

Just to clarify: The current `skip `and `limit`, does not assume sorting, just a mechanism for batching the results. It neither assumes reproducibility (going back to page 1 from page 2 could return different results). Also, when multiple collections are returned, the `limit `value apply to the whole response, not to every collection.
IMHO, the Microsoft guide does an interesting job making explicit the assumptions that API clients SHOULD NOT do.
